finetune/base_models/bert/roberta_encoder.py

```diff
@@ -57,18 +61,84 @@ def _add_extra_toks(self):
         self.mask_token = self.encoder.get("<mask>", 3)
 
 
-
-class RoBERTaEncoderV2(RoBERTaEncoder):
+class RoBERTaEncoderSlow(RoBERTaEncoder):
     """
     A modified wrapper for a public python BPE tokenizer. The modifications allow encoding directly into the formats
     required for finetune. Particularly with respect to formatting with multiple inputs.
-
     Now with support for MLM objective
     """
+
     offset = 4
 
     def _convert_to_embed_idx(self, idx):
         return idx
 
     def __init__(self, encoder_path=ENCODER_PATH, vocab_path=VOCAB_PATH):
         super().__init__(encoder_path=encoder_path, vocab_path=vocab_path)
+
+
+class RoBERTaEncoderV2(BaseEncoder):
+    offset = 4
+
+    def __init__(self, encoder_path=ENCODER_PATH, vocab_path=VOCAB_PATH):
+        self.tokenizer = RobertaTokenizerFast(
+            merges_file=vocab_path, vocab_file=encoder_path
+        )
+        self.start_token = 0  # bos from roberta
+        self.delimiter_token = 2  # eos from roberta
+        self.end_token = 2  # eos from roberta
+        self.UNK_IDX = 3  # unk from roberta
+        self.mask_token = 50264
+        self.mapping = {
+            self.tokenizer.cls_token_id: 0,
+            self.tokenizer.eos_token_id: 2,
+            self.tokenizer.unk_token_id: 3,
+            self.tokenizer.sep_token_id: 2,
+        }
+        self.initialized = True
+
+    @property
+    def vocab_size(self):
+        return 50269
+
+    def _encode(self, texts):
+        batch_tokens = []
+        batch_token_idxs = []
+        batch_char_ends = []
+        batch_char_starts = []
+        for i, text in enumerate(texts):
+            encoded = self.tokenizer._tokenizer.encode(text, add_special_tokens=False)
+            batch_token_idxs.append(
+                [
+                    i + self.offset if i not in self.mapping else self.mapping[i]
+                    for i in encoded.ids
+                ]
+            )
+            token_ends = []
+            token_starts = []
+            tokens = []
+            for (start, end), t in zip(encoded.offsets, encoded.tokens):
+                if t.startswith("Ġ") or t.startswith("Ċ"):
```

Comment:
Is there a reason we exclude these specific chars?

Reply:
RoBERTa tokenizer works on a byte level. My understanding is that when you put these bytes back together and attempt to decode them this is what the space char decodes to. So Indico Data Solutions might tokenize so something like `[Indico ĠData ĠSolutions]`. For this the tokens are used for error checking context alignment on Docrep / LayoutLM etc (and maybe some other places) so we have to strip these so that the resulting tokens align with the input text.

Reply:
ahh thanks for clarifying. I didn't realize that the space char decodes to either `Ġ` or `Ċ`